pkg/apis/pipeline/v1alpha1/verificationpolicy_validation_test.go

```diff
@@ -117,7 +117,74 @@ func TestVerificationPolicy_Invalid(t *testing.T) {
 				},
 			},
 		},
-		want: apis.ErrMultipleOneOf("key[0].data", "key[0].secretref"),
+		want: apis.ErrMultipleOneOf("data", "kms", "secretref").ViaFieldIndex("key", 0),
+	}, {
+		name: "should not have both data and KMS",
+		verificationPolicy: &v1alpha1.VerificationPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "vp",
+			},
+			Spec: v1alpha1.VerificationPolicySpec{
+				Resources: []v1alpha1.ResourcePattern{{".*"}},
+				Authorities: []v1alpha1.Authority{
+					{
+						Name: "foo",
+						Key: &v1alpha1.KeyRef{
+							Data: "inlinekey",
+							KMS:  "kms://key/path",
+						},
+					},
+				},
+			},
+		},
+		want: apis.ErrMultipleOneOf("data", "kms", "secretref").ViaFieldIndex("key", 0),
+	}, {
+		name: "should not have both secretref and KMS",
+		verificationPolicy: &v1alpha1.VerificationPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "vp",
+			},
+			Spec: v1alpha1.VerificationPolicySpec{
+				Resources: []v1alpha1.ResourcePattern{{".*"}},
+				Authorities: []v1alpha1.Authority{
+					{
+						Name: "foo",
+						Key: &v1alpha1.KeyRef{
+							SecretRef: &corev1.SecretReference{
+								Name:      "name",
+								Namespace: "namespace",
+							},
+							KMS: "kms://key/path",
+						},
+					},
+				},
+			},
+		},
+		want: apis.ErrMultipleOneOf("data", "kms", "secretref").ViaFieldIndex("key", 0),
+	}, {
+		name: "should not have data, secretref and KMS at the same time",
+		verificationPolicy: &v1alpha1.VerificationPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "vp",
+			},
+			Spec: v1alpha1.VerificationPolicySpec{
+				Resources: []v1alpha1.ResourcePattern{{".*"}},
+				Authorities: []v1alpha1.Authority{
+					{
+						Name: "foo",
+						Key: &v1alpha1.KeyRef{
+							Data: "inlinekey",
+							SecretRef: &corev1.SecretReference{
+								Name:      "name",
+								Namespace: "namespace",
+							},
+							KMS: "kms://key/path",
+						},
+					},
+				},
+			},
+		},
+		want: apis.ErrMultipleOneOf("data", "kms", "secretref").ViaFieldIndex("key", 0),
```

Comment:
This file may not be the right place, but we should add a test that loads and uses a KMS provider. https://pkg.go.dev/github.com/sigstore/sigstore@v1.5.0/pkg/signature/kms/fake be likely useful here (or feel free to write your own).

Reply:
Yes this PR only adds the filed in api and not really use it. 
Did you suggest add tests to mock kms when we use it?

Reply:
I plan to divide this feature into 3 small PRs, 1 adds libraries, 2 adds field and 3 uses it .
Should I use the `fakekms` to mock in the 3rd pr?